CMakeLists.txt

```diff
@@ -522,6 +522,7 @@ list(APPEND JIT_TEST_SRCS
   ${NVFUSER_ROOT}/tests/cpp/test_swizzle.cpp
   ${NVFUSER_ROOT}/tests/cpp/test_tensor_factories.cpp
   ${NVFUSER_ROOT}/tests/cpp/test_unary.cpp
+  ${NVFUSER_ROOT}/tests/cpp/multidevice.cpp
```

Comment:
IIUC, this line is added because this PR adds some test utils into `multidevice.cpp` that are used for tests running on single-device. 

It may be better to split the file into 
1. `multidevice.*` (or `sharding.*`), which contains test utils that are useful for both single-device and multi-device, and 
2. `multidevice_test.*`, which contains the declaration and definition of `MultiDeviceTest`. 

This way, MultiDeviceTest and its dependencies don't get unnecessarily pulled into `bin/nvfuser_tests` which only run single-device tests. 

Wdyt?

Reply:
I see your point -- keeping separated multi-device test and single device tests. I think bottom-line here it would just mean move `checkSameShardings` away from `multidevice.cpp` (and place it for example in `utils.*`), right?

Reply:
That's right. I suggested the name `multidevice_test.*` because it matches the class name `MultideviceTest`.